pkg/metrics/prometheus_exporter.go

```diff
@@ -29,9 +30,7 @@ func newPrometheusExporter() (view.Exporter, error) {
 	// Start the server for Prometheus scraping
 	go func() {
 		srv := startNewPromSrv(e, *prometheusPort)
-		if err := srv.ListenAndServe(); err != nil {
-			log.Error(err, "Failed to ListenAndServe")
-		}
+		stdlog.Fatal(srv.ListenAndServe())
```

Comment:
Best to not hard kill like this, then the server can't do things like clean up finalizers and old state.

Reply:
makes sense, reverted